src/shared-styles/_logical-properties.scss

```diff
@@ -1 +1,290 @@
+@use "direction";
 @use "rem";
+@use "validation";
+
+@function _maybe-rem($to-rem, $values) {
+  @if ($to-rem == false) {
+    @return $values;
+  } @else {
+    @return rem.calculate($values...);
+  }
+}
+
+@mixin _maybe-rem($to-rem, $properties, $values: ()) {
+  @if ($to-rem == false) {
+    @if type-of($properties) == "map" {
+      @each $property in map-keys($properties) {
+        @include _maybe-rem($to-rem, $property, map-get($properties, $property));
+      }
+    } @else {
+      @each $property in $properties {
+        #{$property}: $values;
+      }
+    }
+  } @else {
+    @include rem.calculate($properties, $values...);
+  }
+}
+
+@function _get-property-prefix($property-name, $dimension) {
+  $physical: (top, right, bottom, left);
+  @if $property-name == inset and index($physical, $dimension) {
+    @return "";
+  }
+  @return "#{$property-name}-";
+}
+
+@function _property-name($parts, $dimension) {
+  @if (length($parts) == 0 or $parts == "") {
+    @return $dimension;
+  }
+
+  @if (length($parts) == 1) {
+    @return "#{_get-property-prefix($parts, $dimension)}#{$dimension}";
+  }
+
+  @return "#{_get-property-prefix(nth($parts, 1), $dimension)}#{$dimension}-#{nth($parts, 2)}";
+}
+
+@mixin property($property-name, $dimension, $arguments, $to-rem: true) {
+  @if length($property-name) > 2 {
+    @include validation.error("Expected at most two property name parts");
+  } @else if $dimension == inline {
+
+    @if type-of($arguments) == "list" and length($arguments) > 1 {
+
+      @include validation.expect-at-most($arguments, 2, "More than two arguments supplied with 'inline' dimension") {
```

Comment:
`@include expect.at-most(` reads quite nicely. 🤔

Reply:
It reads well, but am slightly wary of introducing that as there's already an `expect` mixin provided by the test framework. Similar argument against `assert.at-most` which I also considered. Do you think `validate.expect-at-most` improves on what's currently there?